AirCasting/MicrophoneSession/LevelMeasurementController/Interfaces/LevelSampler.swift

```diff
@@ -0,0 +1,15 @@
+// Created by Lunar on 17/05/2022.
+//
+
+import Foundation
+
+/// Whenever a sampler disconnects this error type is returned
+struct LevelSamplerDisconnectedError: Error { }
+
+/// Interface for objects capable of taking a single measurement
+protocol LevelSampler {
+    associatedtype Measurement
+    /// Takes a single measurement of the `Measurement` type
+    /// - Parameter completion: completion closure called when the measurement is taken or an error is produced
+    func sample(completion: (Result<Measurement, Error>) -> Void)
+}
```

Comment:
As always - it depends. We should always have in mind the perspective of future developers (most likely ourselves in the short term) which will be reading this file. The natural way of reading code is top-down, so we want all prerequisites and assertions on the very top (usually). I think this error type is something we really want to emphasize as it's pretty important for the rest of this module.

So in this form I think the error should be higher than the protocol. But that's a good find! As it's kind of smelly. I'll change it so that the protocol returns a typed error instead of a generic (`Error`).